service/src/test/java/greencity/service/UserServiceImplTest.java

```diff
@@ -234,11 +235,45 @@ void saveTest() {
 
     @Test
     void updateEmployeeEmailTest() {
-        User user = getUser();
-        User updatedUser = getUserWithNewEmail();
-        when(userRepo.findByEmail("taras@gmail.com")).thenReturn(Optional.of(user));
-        userService.updateEmployeeEmail("taras@gmail.com", "test@mail.com");
-        assertEquals(updatedUser.getEmail(), user.getEmail());
+        String uuid = "444e66e8-8daa-4cb0-8269-a8d856e7dd15";
+        String email = "test1@gmail.com";
+        when(userRepo.findUserByUuid(uuid)).thenReturn(Optional.of(user));
+        when(userRepo.existsUserByEmail(email)).thenReturn(false);
+        userService.updateEmployeeEmail(email, uuid);
+        assertEquals(email, user.getEmail());
+        verify(userRepo).findUserByUuid(uuid);
+        verify(userRepo).existsUserByEmail(email);
+    }
+
+    @Test
+    void updateEmployeeWithSameEmailTest() {
+        String uuid = "444e66e8-8daa-4cb0-8269-a8d856e7dd15";
+        String email = "test@gmail.com";
+        when(userRepo.findUserByUuid(uuid)).thenReturn(Optional.of(user));
+        userService.updateEmployeeEmail(email, uuid);
+        assertEquals(email, user.getEmail());
+        verify(userRepo).findUserByUuid(uuid);
+    }
+
+    @Test
+    void updateEmployeeEmailThrowsUsernameNotFoundExceptionTest() {
+        String uuid = "444e66e8-8daa-4cb0-8269-a8d856e7dd15";
+        when(userRepo.findUserByUuid(uuid)).thenReturn(Optional.empty());
+        assertThrows(UsernameNotFoundException.class,
+            () -> userService.updateEmployeeEmail("test@mail.com", uuid));
+        verify(userRepo).findUserByUuid(uuid);
+    }
+
+    @Test
+    void updateEmployeeEmailThrowsUserAlreadyRegisteredExceptionTest() {
+        String uuid = "444e66e8-8daa-4cb0-8269-a8d856e7dd15";
+        String email = "test1@gmail.com";
+        when(userRepo.findUserByUuid(uuid)).thenReturn(Optional.of(user));
+        when(userRepo.existsUserByEmail(email)).thenReturn(true);
+        assertThrows(BadRequestException.class,
```

Comment:
I see in test name `ThrowsUserAlreadyRegisteredException` but here you expect `BadRequestException`

Reply:
fixed

Reply:
done